### ISSUE TITLE
Flashing control board doesn't work from exe (Trac #87)

### ISSUE THREAD

**Issue by cfobel** [wheeler-microfluidics/microdrop#91]
Flashing works fine when program is run through the python interpreter. The problem occurs on both Win 7 and XP.

Migrated from http://microfluidics.utoronto.ca/microdrop/ticket/87
```json
{
    "status": "closed", 
    "changetime": "2014-04-17T19:39:01", 
    "description": "Flashing works fine when program is run through the python interpreter. The problem occurs on both Win 7 and XP.", 
    "reporter": "ryan", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1397763541728826", 
    "component": "microdrop", 
    "summary": "Flashing control board doesn't work from exe", 
    "priority": "blocker", 
    "keywords": "", 
    "version": "0.1", 
    "time": "2012-03-30T03:29:39", 
    "milestone": "Microdrop 1.0", 
    "owner": "ryan", 
    "type": "defect"
}
```

**Comment by cfobel** [comment]
Trac update at 20120330T03:52:39: ryan commented:

> Update: Flashing works fine from a Win 7 exe if the exe is compiled on Win 7. So the problem has something to do with the exe compiled on XP.
> 
> Flashing fails due to the following exception:
> 
> Traceback (most recent call last):
>   File "C:\buildbot\microdrop\WinXP_Py2_7\build\build\pyi.win32\microdrop\outPYZ
> 1.pyz/logging", line 842, in emit
>   File "C:\buildbot\microdrop\WinXP_Py2_7\build\build\pyi.win32\microdrop\outPYZ
> 1.pyz/logging", line 719, in format
>   File "C:\buildbot\microdrop\WinXP_Py2_7\build\build\pyi.win32\microdrop\outPYZ
> 1.pyz/logging", line 464, in format
>   File "C:\buildbot\microdrop\WinXP_Py2_7\build\build\pyi.win32\microdrop\outPYZ
> 1.pyz/logging", line 328, in getMessage
> TypeError: not all arguments converted during string formatting
> Logged from file __init__.py, line 201

**Comment by cfobel** [comment]
Trac update at 20120330T16:35:26: ryan commented:

> Correction, flashing works fine from the distribution directory, but not from the msi installed version. The behaviour is the same on Win7 and XP. It is caused by looking for avrdude.exe in the wrong place:
> 
> avrdude path=C:\Program Files (x86)\microdrop\plugins\dmf_control_board\avr\avrdude.exe
> 
> when actually, it is in the plugins directory.

**Comment by cfobel** [comment]
Trac update at 20120330T16:48:53: 
* ryan@fobel.net commented:

> commit 44f981c265f7c58bfe2ae72044f13b13ca07c783
> Author: Ryan Fobel <ryan@fobel.net>
> Date:   Fri Mar 30 12:46:15 2012 -0400
> 
>     [SUBMODULE] Update dmf_control_board
>     
>     Fixes #91.


* ryan@fobel.net changed resolution from "" to "fixed"
* ryan@fobel.net changed status from "new" to "closed"

**Comment by cfobel** [comment]
Trac update at 20140417T19:39:01: ryan changed component from "hardware/communication" to "microdrop"
